src/instruction.rs

```diff
@@ -363,38 +459,40 @@ pub enum InstructionRole {
 impl From<&Instruction> for InstructionRole {
     fn from(instruction: &Instruction) -> Self {
         match instruction {
-            Instruction::CalibrationDefinition(_)
-            | Instruction::CircuitDefinition { .. }
-            | Instruction::Declaration { .. }
-            | Instruction::FrameDefinition { .. }
-            | Instruction::Gate { .. }
-            | Instruction::GateDefinition { .. }
-            | Instruction::Label(_)
-            | Instruction::MeasureCalibrationDefinition { .. }
-            | Instruction::Measurement { .. }
-            | Instruction::Pragma { .. }
-            | Instruction::WaveformDefinition { .. } => InstructionRole::ProgramComposition,
-            Instruction::Reset { .. }
-            | Instruction::Capture { .. }
-            | Instruction::Delay { .. }
-            | Instruction::Fence { .. }
-            | Instruction::Pulse { .. }
-            | Instruction::RawCapture { .. }
-            | Instruction::SetFrequency { .. }
-            | Instruction::SetPhase { .. }
-            | Instruction::SetScale { .. }
-            | Instruction::ShiftFrequency { .. }
-            | Instruction::ShiftPhase { .. }
-            | Instruction::SwapPhases { .. } => InstructionRole::RFControl,
-            Instruction::Arithmetic { .. }
-            | Instruction::Move { .. }
-            | Instruction::Exchange { .. }
-            | Instruction::Load { .. }
+            Instruction::CalibrationDefinition(CalibrationDefinition(_))
+            | Instruction::CircuitDefinition(CircuitDefinition { .. })
```

Comment:
If ignoring the interior, couldn't you still have this for brevity?

```suggestion
            Instruction::CalibrationDefinition(_)
            | Instruction::CircuitDefinition(_)
```